mods/ra/maps/allies-10a/allies10a.lua

```diff
@@ -0,0 +1,220 @@
+		Media.PlaySpeechNotification(Greece, "AbombLaunchDetected")
+		MissileSilo4.ActivateNukePower(CPos.New(127, 127))
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(8), function()
+		local fmvStart = DateTime.GameTime
+		Media.PlayMovieFullscreen("ally10b.vqa", function()
+			-- Completing immediately indicates that the FMV is not available
+			-- Fall back to a text message
+			if fmvStart == DateTime.GameTime then
+				Media.DisplayMessage("Commander, we're tracking four missiles. They must be deactivated! We are scrambling a team to assult the missile control bunker. Clear the way and capture the enemy command center. Hurry!", "LANDCOM 16")
+			end
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(9), function()
+		CaptureFCom = Greece.AddObjective("Capture the enemy Command Center. Hurry!")
+		StartTimer = true
+		Media.PlaySpeechNotification(Greece, "TimerStarted")
+		Greece.MarkCompletedObjective(ApproachBase)
+	end)
+
+	Trigger.OnCapture(CommandCenter, function()
+		Greece.MarkCompletedObjective(CaptureFCom)
+	end)
+
+	Paradrop()
+	Trigger.AfterDelay(DateTime.Minutes(2), SendParabombs)
+end
+
+SilosDamaged = function()
+	if not MissilesLaunched then
+		LaunchMissiles()
+	end
+end
+
+Tick = function()
+	USSR.Cash = 50000
+	BadGuy.Cash = 50000
+
+	if StartTimer then
+		if ticked > 0 then
+			UserInterface.SetMissionText("Missiles reach their targets in " .. Utils.FormatTime(ticked), TimerColor)
+			ticked = ticked - 1
+		elseif ticked == 0 then
+			UserInterface.SetMissionText("We're too late!", USSR.Color)
+			USSR.MarkCompletedObjective(HoldOut)
+		end
+	end
+
+	if Greece.HasNoRequiredUnits() then
+		USSR.MarkCompletedObjective(HoldOut)
+	end
+end
+
+WorldLoaded = function()
+	Greece = Player.GetPlayer("Greece")
+	USSR = Player.GetPlayer("USSR")
+	BadGuy = Player.GetPlayer("BadGuy")
+
+	Trigger.OnObjectiveAdded(Greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	HoldOut = USSR.AddObjective("Hold out until missiles reach their destination")
+	ApproachBase = Greece.AddObjective("Find a way to take the atomic weapons off-line.")
+
+	Trigger.OnObjectiveCompleted(Greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(Greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+	Trigger.OnPlayerLost(Greece, function()
+		Media.PlaySpeechNotification(Greece, "Lose")
+	end)
+	Trigger.OnPlayerWon(Greece, function()
+		Media.PlaySpeechNotification(Greece, "Win")
+	end)
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	StandardDrop = Actor.Create("paradrop", false, { Owner = USSR })
+	MissionStart()
+	MissionTriggers()
+	ActivateAI()
+	OnAnyDamaged(MissileSilos, SilosDamaged)
+
+	Trigger.AfterDelay(DateTime.Minutes(30), function()
+		Media.PlaySpeechNotification(Greece, "ThirtyMinutesRemaining")
+	end)
```

Comment:
First of all, please use the TimeLimitManager here as well. Second of all, what's the point of these notifications? As far as I can see, nothing special is going to happen at 60 minutes into the game. Were the nukes supposed to launch automatically then?

Reply:
I've got timer notifications in the wrong place. They are meant to match the timer after the nukes launch.
Does `TimeLimitManager` work with timers that start midway through a mission?

Reply:
> Does TimeLimitManager work with timers that start midway through a mission?

Yes.